File: src/test/java/com/raychenon/leetcode/PermutationinStringTest.java
package com.raychenon.leetcode;

import org.junit.BeforeClass;
import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 19/5/2020
 * https://leetcode.com/problems/permutation-in-string/
 */
public class PermutationinStringTest {

    private static PermutationinString permutationinString;

    @BeforeClass
    public static void setUp() {
        permutationinString = new PermutationinString();
    }

    @Test
    public void example1Test() {
        assertPermutationinStringMethods("ab", "eidbaooo", true);
    }

    @Test
    public void example2Test() {
        assertPermutationinStringMethods("ab", "eidboaoo", false);
    }

    @Test
    public void example3Test() {
        assertPermutationinStringMethods("acb", "ooeabidabcoaoo", true);
    }

    private void assertPermutationinStringMethods(String s1, String s2, boolean expected) {
        assertEquals(permutationinString.checkInclusion(s1, s2), expected);
    }

}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/PermutationinStringTest.kt
package com.raychenon.kotlin.leetcode

import junit.framework.TestCase
import org.junit.Test

/**
 * User: raychenon
 * Date: 19/5/2020
 * https://leetcode.com/problems/permutation-in-string
 */
class PermutationinStringTest {

    @Test
    fun example1Test() {
        assertPermutationinStringMethods("ab", "eidbaooo", true)
    }

    @Test
    fun example2Test() {
        assertPermutationinStringMethods("ab", "eidboaoo", false)
    }

    @Test
    fun example3Test() {
        assertPermutationinStringMethods("acb", "ooeabidabcoaoo", true)
    }

    private fun assertPermutationinStringMethods(
        s1: String,
        s2: String,
        expected: Boolean
    ) {
        TestCase.assertEquals(
            PermutationinString.checkInclusion(s1, s2),
            expected
        )
    }
}